fqdn/__init__.py

```diff
@@ -34,10 +34,14 @@ class FQDN:
     def __init__(self, fqdn):
         if not (fqdn and isinstance(fqdn, str)):
             raise ValueError("fqdn must be str")
-        self.fqdn = fqdn
+        self._fqdn = fqdn
 
     def __str__(self):
-        return self.fqdn
+        return self._fqdn
+
+    @property
+    def fqdn(self):
+        return self._fqdn
```

Comment:
However, `self.fqdn` was never mentioned in the [docs](https://fqdn.readthedocs.io/en/latest/).  The docs are the  API contract. Someone who went in to look at the source to learn about `self.fqdn` ought to have looked close enough to see the `cached_property` decorator. Even if they hadn't, the docs only show properties, no methods; there should have been no expectation that changing the string in `self.fqdn` would change `self.absolutes` or `self.relative`. Therefore I see no reason to keep `self.fqdn` because the API contract remains the same.

To be nice, I'll do a minor version bump instead of just a patch.

Reply:
awesome, that makes much more sense. Just had libraries push back in the past ;)